src/newtondynamics_physics/SysNewton.cpp

```diff
@@ -0,0 +1,428 @@
+/**
+ * Open Space Program
+ * Copyright © 2019-2020 Open Space Program Project
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "SysNewton.h"
+
+#include <osp/Active/SysPhysics.h>
+#include <osp/Active/SysHierarchy.h>
+
+#include <Newton.h>
+
+using ospnewton::SysNewton;
+using ospnewton::ACompNwtBody;
+using ospnewton::ACompNwtWorld;
+
+using osp::active::ActiveScene;
+using osp::active::ActiveReg_t;
+using osp::active::ActiveEnt;
+
+using osp::active::SysPhysics;
+
+using osp::active::ACompHierarchy;
+using osp::active::ACompRigidBody;
+using osp::active::ACompShape;
+using osp::active::ACompSolidCollider;
+using osp::active::ACompTransform;
+using osp::active::ACompTransformControlled;
+using osp::active::ACompTransformMutable;
+
+using osp::Matrix4;
+using osp::Vector3;
+
+// Callback called for every Rigid Body (even static ones) on NewtonUpdate
+void cb_force_torque(const NewtonBody* pBody, dFloat timestep, int threadIndex)
+{
+    // Get Scene from Newton World
+    ActiveScene* scene = static_cast<ActiveScene*>(
+                NewtonWorldGetUserData(NewtonBodyGetWorld(pBody)));
+
+    // Get Associated entity
+    uintptr_t userData = reinterpret_cast<uintptr_t>(NewtonBodyGetUserData(pBody));
+    ActiveEnt ent = static_cast<ActiveEnt>(userData);
+
+    auto &rBody = scene->reg_get<ACompRigidBody>(ent);
+    auto &rTransformComp = scene->reg_get<ACompTransform>(ent);
+
+    // Check if transform has been set externally
+    auto &tfMutable = scene->reg_get<ACompTransformMutable>(ent);
+    if (tfMutable.m_dirty)
```

Comment:
minor:

if(std::exchange(tfMutable.m_dirty, false))
{
    NewtonBodySetMatrix(...);
}

if you care for that style.